src/sparseml/pytorch/optim/manager.py

```diff
@@ -281,14 +284,25 @@ def from_yaml(
         recipe_variables = parse_recipe_variables(recipe_variables)
         yaml_str = load_recipe_yaml_str(file_path, **recipe_variables)
         modifiers = Modifier.load_list(yaml_str)
-
         if add_modifiers:
             modifiers.extend(add_modifiers)
 
-        metadata = validate_metadata(metadata, yaml_str)
+        validated_metadata = validate_metadata(metadata, yaml_str)
 
-        manager = ScheduledModifierManager(modifiers=modifiers, metadata=metadata)
+        if metadata is not None:
+            logging.info(
+                "The new `metadata` has been passed to the manager. "
+                f"This will change the metadata in the recipe {file_path}"
+                f"including updating the metadata key {FRAMEWORK_METADATA_KEY} "
+                "to reflect your current setup."
+            )
```

Comment:
instead of this log, let's just do an info log after the metadata is complete that prints out the entire metadata.  If new metadata is passed from the from yaml constructor, then it likely won't have the framework metadata key so does not make sense to log that we are updating it

Reply:
let's also do this for the other framework managers.